js/src/javascript/beautifier.js

```diff
@@ -973,11 +968,9 @@ Beautifier.prototype.handle_word = function(current_token) {
   }
 
   if (reserved_array(current_token, ['else', 'catch', 'finally'])) {
-    if ((!(this._flags.last_token.type === TOKEN.END_BLOCK && this._previous_flags.mode === MODE.BlockStatement) ||
-        this._options.brace_style === "expand" ||
-        this._options.brace_style === "end-expand" ||
-        (this._options.brace_style === "none" && current_token.newlines)) &&
-      !this._flags.inline_frame) {
+    var blockStatementEnds = this._flags.last_token.type === TOKEN.END_BLOCK && this._previous_flags.mode === MODE.BlockStatement;
+    if ((!blockStatementEnds || in_array(this._options.brace_style, ["expand", 'end-expand', 'expand-all']) ||
```

Comment:
New feature to have same behavior as that of brace style option with value expand or end-expand.